data_model/src/block.rs

```diff
@@ -167,32 +167,56 @@ impl SignedBlockV1 {
 }
 
 impl SignedBlock {
-    /// Block payload
-    // FIXME: Leaking concrete type BlockPayload from Versioned container. Payload should be versioned
-    pub fn payload(&self) -> &BlockPayload {
+    /// Block transactions
+    pub fn transactions(&self) -> impl ExactSizeIterator<Item = &TransactionValue> {
         let SignedBlock::V1(block) = self;
-        block.payload()
+        block.payload.transactions.iter()
+    }
+
+    /// Block header
+    #[inline]
+    pub fn header(&self) -> &BlockHeader {
```

Comment:
do you think we should expose that blocks have headers? or should we do the same as for the `payload`?

Reply:
There are 18 places where the headers are used. If we hide the method, we should define a versioned SignedBlock on each of these place. And my thoughts hint me that it would influence on the performance and the convenience of the code maintenance.

Reply:
what I meant was this: instead of having `block.header().height` we can have `block.height()`. Likewise for all other header fields

Reply:
It's uses in few places, so I decided to keep it.
But the getters of else BlockHeader fields I've implemented in the SignedBlock

Reply:
>  But the getters of else BlockHeader fields I've implemented in the SignedBlock

please revert that

Reply:
Reverted all getters back. Kept headers() only